lib/mist/s3/buckets.py

```diff
@@ -37,21 +38,41 @@ def handle_delete(self):
         try:
             patterns = [re.compile(arg) for arg in args[1:]]
             buckets = self.find_matching(patterns)
-            bucket_names = [{"bucket": bucket.name} for bucket in buckets]
             try:
-                while len(buckets) > 0:
-                    bucket = buckets[0]
+                for bucket in buckets:
                     bucket.delete()
-                    buckets = buckets[1:]
             except Exception as e:
-                Logger.error("Error deleting S3 bucket '%s': %s" & (buckets[0].name, e))
+                Logger.error("Error deleting S3 bucket '%s': %s" % (buckets[0].name, e))
                 self.resp.send_error("Error deleting bucket '%s': %s" % (buckets[0].name, e))
             if len(buckets) == 0:
                 self.resp.append_body([], template="empty_result")
             else:
-                self.resp.append_body({"buckets": buckets}, template="delete_buckets")
+                bucket_names = [bucket.name for bucket in buckets]
+                self.resp.append_body({"buckets": ", ".join(bucket_names)}, template="delete_buckets")
         except Exception as e:
-            self.resp.send_error("One of the following regular expressions is invalid: %s" % (args[1:]))
+            self.resp.send_error("One of the following regular expressions is invalid: %s, %s" % (args[1:], e))
+
+    def handle_create(self):
+        self.handle_new()
+
+    def handle_new(self):
+        if self.req.arg_count() < 2:
+            self.usage_error()
```

Comment:
Do you need a guard here to return from this function? My python is rusty but it looks like it will continue to execute after sending an error response.